include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -1109,13 +1104,23 @@ common::internal::vector<pika::shared_future<common::internal::vector<T>>> Reduc
   // Reflector of size = 1 is not considered whatever T is (i.e. neither real nor complex)
   const SizeType nrefls = std::max<SizeType>(0, dist.size().rows() - band_size - 1);
 
-  common::internal::vector<pika::shared_future<common::internal::vector<T>>> taus;
+  // Row-vector that is distributed over columns, but exists locally on all rows of the grid
+  // TODO: transpose into column vector?
+  DLAF_ASSERT(mat_a.blockSize().cols() % band_size == 0, mat_a.blockSize().cols(), band_size);
+  Matrix<T, Device::CPU> mat_taus(matrix::Distribution(GlobalElementSize(1, nrefls),
+                                                       TileElementSize(1, mat_a.blockSize().cols()),
+                                                       comm::Size2D(1, mat_a.commGridSize().cols()),
+                                                       comm::Index2D(0, mat_a.rankIndex().col()),
+                                                       comm::Index2D(0, mat_a.sourceRankIndex().col())));
 
   if (nrefls == 0)
-    return taus;
+    return mat_taus;
 
-  const SizeType nblocks = (nrefls - 1) / band_size + 1;
-  taus.reserve(nblocks);
+  matrix::RetiledMatrix<T, Device::CPU> mat_taus_retiled(
+      mat_taus, LocalTileSize(1, mat_a.blockSize().cols() / band_size));
```

Comment:
This could become easier once #905 is done. It would allow us to use a `Matrix<...>&` to either the original unretiled matrix or to the retiled matrix since the types would be the same.

Reply:
#910